.github/workflows/build-and-test.yml

```diff
@@ -8,64 +8,30 @@ on:
 
 jobs:
   build-and-test-osx:
-    runs-on: macos-latest
+    runs-on: ${{ matrix.os }}
```

Comment:
AWESOME!! I had been meaning to do this for ages, but never got around to figuring out what the syntax was.

Reply:
I have been making some mental foo of late, you can generate matrices dynamically in your actions, so I have a run of pact ffi/ pact verifier cli and pact mock server cli, for all the targets, in one super clean action.

I just wish you could report on individual matrix combos, because you could build up an awesome compat matrix, with github badges. 

The other way is to have separate workflows , I need to check if you can pull in one master workflow and then have loads of mini workflows, that just specify x combo (node 14 / windows) so they appear as seperate workflows, and you could get a status badge for each one.

Reply:
> I just wish you could report on individual matrix combos, because you could build up an awesome compat matrix, with github badges.

I dunno - I think it's better to just report the whole success / failure - as then you don't get tempted to continue with "well, it doesn't work on everything we deploy to".

It would be nice to be able to generate the list of tested combinations, though!

Reply:
a whole just fail isn’t helpful for me finding the needle in the haystack and github’s view of multi matrix runs needs some work, detail gets lots and you just have 12 boxes that al look the same from the truncated description until you click on one, and that is an expensive use of time

i need to check out github blocks again as there is meant to be some cool stuff you can do in readmes now